content/articles/graphs-in-data-structure -using _c++/index.md

```diff
@@ -0,0 +1,153 @@
+### Graphs in Data structure ( Using C++)
+
+#### Introduction
+Graphs are mathematical structures that reflect the pairwise relationship between things. A graph is a type of flow structure that displays the interactions of several objects. It may be represented by utilizing the two fundamental components shown below:
+
+* **Nodes**: These are the most crucial elements of every graph. Edges are used to represent node connections. A graph that have two nodes connected to each other using an undirected edge shows a bi-directional connection between those two nodes.
+
+* **Edges**: Edges are part of a graph showing the connections between nodes. An edge represent the connection between two nodes.
+
+#### An application in real life 
+
+* **Google Maps**: link your journey from the start to the end
+* **Social Networks**: Friends with each other are connected using an edge where each user represents a vertex.
+* **Recommendation System**: relationship data between users recommendation is use graph for connection.
+
+#### Type of Graphs :
+* **Undirected**: All edges in an undirected graph are bidirectional and do not point in any particular direction.
+
+![graph](./graph_1.png)
+
+* **Directed**: In a directed graph all of the edges are uni-directional they point to a single direction.
+
+![graph](./graph__2.JPG)
+
+* **Weighted**: Weighted graph, with each edge, allocated a weight or cost. Consider a network with four nodes, shown in the picture. Every edge of a graph have some weight with it, if you wish to get from vertex 2 to vertex 1, you can choose one of three routes:
+    * 2->3->1
+    * 2->1
+    * 2->3->4->1
+  
+As a result, the total cost of each path is as follows:
+
+The total cost of 2->3->1 is (2+1), or three units.
+
+The total cost of getting from 2->1 is one unit.
+
+The total cost of 2->3->4->1 is (2+2+3), or seven units.
+
+![graph](./graph_3.JPG)
+
+* **Cyclic**: If a graph contains a path that starts and finishes at the same vertex, it is said to be cyclic. This is referred to as a cycle. An acyclic graph does not have a cycle.
+
+![graph](./graph_4.jpg)
+
+The diagram below depicts a cycle (0->1->2).
+A tree is a graph that has just one path connecting any two vertices. Every node in a graph may have one or more parents. In a tree as each node has precisely one parent node.
+
+A tree cannot have any loops or cycles, whereas graphs may.
+
+![graph](./graph_5.jpg)
+
+### Graph Representation 
+A graph can be represented in several ways. The following are the two most frequent ways of expressing a graph:
```

Comment:
Change "A graph can be represented in several ways. The following are the two most frequent ways of expressing a graph:" to " We can represent a graph in several ways. The following are the two most frequent ways of expressing a graph:"